src/Text/Layout/Table/Primitives/Table.hs

```diff
@@ -10,32 +13,38 @@ import           Text.Layout.Table.Spec.Util
 
 
 -- | Draw a horizontal line that will use the delimiters around the
--- appropriately and visually separate by 'hSpace'.
+-- appropriately and visually separate by 'hSpace', or Nothing if
+-- all delimiters are null.
 hLineDetail
     :: StringBuilder b
-    => Char -- ^ The space character that is used as padding.
-    -> Char -- ^ The delimiter that is used on the left side.
-    -> Char -- ^ The delimiter that is used in between cells.
-    -> Char -- ^ The delimiter that is sued on the right side.
+    => String -- ^ The space characters that are used as padding.
+    -> String -- ^ The delimiter that is used on the left side.
+    -> String -- ^ The delimiter that is used in between cells.
+    -> String -- ^ The delimiter that is used on the right side.
     -> Row b -- ^ A row of builders.
-    -> b -- ^ The formatted line as a 'StringBuilder'.
-hLineDetail hSpace delimL delimM delimR cells =
-    mconcat $ intersperse (charB hSpace) $ charB delimL : intersperse (charB delimM) cells ++ [charB delimR]
+    -> Maybe b -- ^ The formatted line as a 'StringBuilder', or Nothing if all delimiters are null.
+hLineDetail ""     ""     ""     ""     = const Nothing
+hLineDetail hSpace delimL delimM delimR = Just . hLineDetail' hSpace delimL delimM delimR
 
--- | A simplified version of 'hLineDetail' that will use the same delimiter
--- for everything.
-hLine
+-- | Draw a horizontal line that will use the delimiters around the
+-- appropriately and visually separate by 'hSpace'.
+hLineDetail'
     :: StringBuilder b
-    => Char -- ^ The space character that is used as padding.
-    -> Char -- ^ The delimiter that is used for everything.
+    => String -- ^ The space characters that are used as padding.
+    -> String -- ^ The delimiter that is used on the left side.
+    -> String -- ^ The delimiter that is used in between cells.
+    -> String -- ^ The delimiter that is used on the right side.
     -> Row b -- ^ A row of builders.
     -> b -- ^ The formatted line as a 'StringBuilder'.
-hLine hSpace delim = hLineDetail hSpace delim delim delim
+hLineDetail' hSpace delimL delimM delimR cells = mconcat . intersperse (stringB hSpace) $
+    stringB delimL : intersperse (stringB delimM) cells ++ [stringB delimR]
 
 -- | Render a line with actual content.
 hLineContent
     :: StringBuilder b
-    => Char -- ^ The delimiter that is used for everything.
+    => String -- ^ The delimiter that is used on the left side.
+    -> String -- ^ The delimeter that is used in between cells.
+    -> String -- ^ The delimeter that is used on the right side.
     -> Row b -- ^ A row of builders.
     -> b
-hLineContent = hLine ' '
+hLineContent = hLineDetail' " "
```

Comment:
Could we alter the naming slightly?

* `hLineContent` to `horizontalContentLine`
* `hLineDetail'` to `horizontalDetailLine`
* `hLineDetail` to `optHorizontalDetailLine`

What I would like to achieve with this is to bring out the differences in those definitions a bit.  If you have some idea yourself, I am happy to hear about it.

Reply:
Sure. I currently don't export `hLineDetail'`, but I suspect you'd like it included in the export list as well?